analytics/src/main/kotlin/uk/govuk/app/analytics/ui/AnalyticsConsentScreen.kt

```diff
@@ -0,0 +1,186 @@
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Icon
+import androidx.compose.material3.adaptive.currentWindowAdaptiveInfo
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.window.core.layout.WindowHeightSizeClass
+import uk.govuk.app.analytics.AnalyticsViewModel
+import uk.govuk.app.analytics.R
+import uk.govuk.app.design.ui.component.BodyRegularLabel
+import uk.govuk.app.design.ui.component.HorizontalButtonGroup
+import uk.govuk.app.design.ui.component.LargeTitleBoldLabel
+import uk.govuk.app.design.ui.component.ListDivider
+import uk.govuk.app.design.ui.component.VerticalButtonGroup
+import uk.govuk.app.design.ui.theme.GovUkTheme
+
+@Composable
+internal fun AnalyticsConsentRoute(
+    onPrivacyPolicyClick: () -> Unit,
+    analyticsConsentCompleted: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val viewModel: AnalyticsViewModel = hiltViewModel()
+
+    AnalyticsConsentScreen(
+        onPrivacyPolicyClick = onPrivacyPolicyClick,
+        onConsentGranted = {
+            viewModel.onConsentGranted()
+            analyticsConsentCompleted()
+        },
+        onConsentDenied = {
+            viewModel.onConsentDenied()
+            analyticsConsentCompleted()
+        },
+        modifier = modifier
+    )
+}
+
+@Composable
+private fun AnalyticsConsentScreen(
+    onPrivacyPolicyClick: () -> Unit,
+    onConsentGranted: () -> Unit,
+    onConsentDenied: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier
+            .fillMaxWidth()
+    ) {
+        Column(
+            Modifier
+                .weight(1f)
+                .verticalScroll(rememberScrollState())
+                .padding(horizontal = GovUkTheme.spacing.medium)
+                .padding(top = GovUkTheme.spacing.medium)
+        ) {
+            LargeTitleBoldLabel(stringResource(R.string.analytics_consent_title))
+            Spacer(Modifier.height(GovUkTheme.spacing.medium))
```

Comment:
Should we think about creating components for these? Something like...

- SpacerMedium()
- SpacerSmall()

Reply:
Good idea!

Reply:
Perfect - thank you 👍